readthedocs/proxito/cache.py

```diff
@@ -0,0 +1,65 @@
+import structlog
+
+log = structlog.get_logger(__name__)
+
+CACHE_TAG_HEADER = "Cache-Tag"
+CDN_CACHE_CONTROL_HEADER = "CDN-Cache-Control"
+
+
+def add_cache_tags(response, cache_tags):
+    """
+    Add cache tags to the response.
+
+    New cache tags will be appended to the existing ones.
+
+    :param response: The response to add cache tags to.
+    :param cache_tags: A list of cache tags to add to the response.
+    """
+    cache_tags = cache_tags.copy()
+    current_cache_tag = response.headers.get(CACHE_TAG_HEADER)
+    if current_cache_tag:
+        cache_tags.append(current_cache_tag)
+
+    response.headers[CACHE_TAG_HEADER] = ",".join(cache_tags)
+
+
+def cache_response(response, cache_tags=None, force=True):
+    """
+    Cache the response at the CDN level.
+
+    We add the ``Cache-Tag`` header to the response, to be able to purge
+    the cache by a given tag. And we set the ``CDN-Cache-Control`` header
+    to public, to cache the response at the CDN level only. This doesn't
+    affect caching at the browser level (``Cache-Control``).
+
+    See:
+
+    - https://developers.cloudflare.com/cache/how-to/purge-cache/#cache-tags-enterprise-only
+    - https://developers.cloudflare.com/cache/about/cdn-cache-control.
+
+    :param response: The response to cache.
+    :param cache_tags: A list of cache tags to add to the response.
+    :param force: If ``True``, the header will be set to public even if it
+     was already set to private.
+    """
+    if cache_tags:
+        add_cache_tags(response, cache_tags)
+    if force or CDN_CACHE_CONTROL_HEADER not in response.headers:
+        if not response.headers.get(CACHE_TAG_HEADER):
+            log.warning("Caching response without cache tags.")
```

Comment:
Is this a real case or we are logging this here just for debugging purposes?

Reply:
It doesn't make sense to cache a response if we don't have a way to purge it using cache tags.

Reply:
This would be good as a comment, or even better in the logging message.